### ISSUE TITLE
Ingress upstream doesn't refresh after node failure

### ISSUE THREAD

**Issue by boris-savic** [rancher/rancher#26494]
**What kind of request is this (question/bug/enhancement/feature request):** bug

**Steps to reproduce (least amount of steps as possible):**

1. Setup a 3 node cluster where each node has all roles (worker, etcd, control plane)
2. Deploy standard *nginx:alpine* image with scale 1 for a sample http app (any other will do as well) and configure load-balancer with a domainname to the workload/service
3. Check which node has the pod running on it and shutdown the node in your cloud provider - simulate a hard node failure
4. Wait few minutes (untill the cluster detects the node is unreachable and pod status becomes unknown - 5min later it should reschedule pod on a working node)

**Result:**
Very inconsistent, but sometimes after performing the steps above (every so often - like 1 out of 5) the hostname http request keeps returning 504 timeout (there is no 503 when the Pod switches to state *Unknown*) and then once the new pod is created on a working node the ingress still returns 504 code. Nothing at this stage helps - not even increasing the scale up - for example 3 replicas. They get created but ingress (nginx) still returns 504. 

I've checked the Ingress logs and it only requests from upstream IP of the original pod that was on the node we shut down. Even redeploying the workload, reconfiguring the load balancer etc. never updates the service ip that the Ingress has in the upstream list. 

Basically only way to make the service work again is to reconnect the node that failed. 

**Other details that may be helpful:**

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): 2.3.6
- Installation option (single install/HA): HA

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported): Custom
- Machine type (cloud/VM/metal) and specifications (CPU/memory): Hetzner Cloud - 8GB RAM
- Kubernetes version (use `kubectl version`): v1.17.4-rancher1-2

- Docker version (use `docker version`): 1.17.4

**Comment by boris-savic** [comment]
After some more testing I've noticed that this happens when the **cattle-cluster-agent** pod is scheduled on the node that is downed.
